### PR TITLE
[EHT-592] Sui validator setup

### PR DESCRIPTION
**Pull request by kowshikvajrala**

### CODE REVIEW

File: roles/node/templates/mainnet/validator/config.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    - address: /dns/sjc-00.mainnet.sui.io/udp/8084
+      peer-id: 6f0b25087cd6b2fd2e4329bcf308ac95a37c49277dd7286b72470c124809db5b
+    - address: /dns/sjc-01.mainnet.sui.io/udp/8084
+      peer-id: af1d5d8468b3612ac2b6ff3ca91e99a71390dbe5b83dea9f6ae2da708d689227
+    - address: /dns/lhr-00.mainnet.sui.io/udp/8084
+      peer-id: c619a5e0f8f36eac45118c1f8bda28f0f508e2839042781f1d4a9818043f732c
+    - address: /dns/lhr-01.mainnet.sui.io/udp/8084
+      peer-id: 53dcedf250f73b1ec83250614498947db00d17c0181020fcdb7b6db12afbc175
+  listen-address: 0.0.0.0:8084
+  external-address: /dns/{{ sui.validator.external_address }}/udp/8084
```
Comment (aditya-manit) [inline roles/node/templates/mainnet/validator/config.yaml.j2:66]: please store the port in group var

File: roles/node/templates/mainnet/validator/config.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,94 @@
+---
+protocol-key-pair:
+  path: "{{ node_protocol_key_path }}"
+worker-key-pair:
+  path: "{{ node_worker_key_path }}"
+network-key-pair:
+  path: "{{ node_network_key_path }}"
+db-path: "{{ node_authorities_data_dir }}"
+network-address: /ip4/0.0.0.0/tcp/8080/http
+json-rpc-address: 0.0.0.0:9000
```
Comment (aditya-manit) [inline roles/node/templates/mainnet/validator/config.yaml.j2:10]: This needs to be opened as well

File: roles/node/vars/ufw_rules.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+node_ufw_rules:
+  validator:
```
[the remark this answers is not in the document]
Comment (aditya-manit) [reply roles/node/vars/ufw_rules.yml:3]: Please check once if they have anything on ports in their official docs

File: roles/node/templates/testnet/fullnode/config.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Update this value to the location you want Sui to store its database
+db-path: "{{ node_data_dir }}"
+
+# For ipv4, update this to "/ip4/X.X.X.X/tcp/8080/http"
+network-address: "/dns/localhost/tcp/8080/http"
+metrics-address: "0.0.0.0:{{ sui.fullnode.ports.metrics_port }}"
+# this address is also used for web socket connections
+json-rpc-address: "0.0.0.0:9000"
+enable-event-processing: true
```
Comment (aditya-manit) [inline roles/node/templates/testnet/fullnode/config.yaml.j2:9]: should be fetched from group vars

**Review by aditya-manit — CHANGES_REQUESTED** [review]
Port situations needs to be checked again

1. Please check configs again and fetch ports from group var, right now some of them are being are while other arent
2. Some ports requires only udp traffic while other tcp
3. Please check their docs if they have added anything on ports and ufw rules
4. Update services if needed from https://www.notion.so/encapsulate/Services-15c14f565a5080aa930df5ea6e550f19?pvs=4

File: roles/node/templates/testnet/systemd.service.j2 [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Sui node
+
+[Service]
+User={{ service_identifier }}
+WorkingDirectory={{ node_home_dir }}
+Environment=RUST_BACKTRACE=1
+Environment=RUST_LOG=info,sui_core=debug,narwhal=debug,consensus=debug,jsonrpsee=error
+ExecStart={{ node_bin_dir }}/{{ service_identifier }} --config-path {{ node_config_file_path }}
```
[the remark this answers is not in the document]
Comment (aditya-manit) [reply roles/node/templates/testnet/systemd.service.j2:9]: https://www.notion.so/encapsulate/Services-15c14f565a5080aa930df5ea6e550f19?pvs=4

**Comment by aditya-manit** [comment]
I am merging this PR, lets address the comments in another bracnch

File: roles/node/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    - name: Fetch network key and save
+      ansible.builtin.include_tasks: tasks/get_and_save_vault_secret.yml
+      vars:
+        vault_secret_source_path: "{{ project }}/{{ org }}/{{ type }}/{{ node_network_key_file_name }}"
+        secret_destination_path: "{{ node_network_key_path }}"
+        secret_type: key
+
+- name: Fetch the service binary
+  ansible.builtin.get_url:
```
[the remark this answers is not in the document]
Comment (kowshikvajrala) [reply roles/node/tasks/main.yml:62]: The name of sui cli binary is sui-bridge-cli ?

**Pull request merged**